lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex

```diff
@@ -181,20 +180,10 @@ defmodule LambdaEthereumConsensus.ForkChoice do
   def recompute_head(store) do
     {:ok, head_root} = Helpers.get_head(store)
 
-    head_block = Blocks.get_block!(head_root)
-    head_execution_hash = head_block.body.execution_payload.block_hash
+    Handlers.notify_forkchoice_update(store, head_root)
 
+    head_block = Blocks.get_block!(head_root)
```

Comment:
We should also pass the `head_block`, to avoid fetching the same block twice

Reply:
Good call!